Sources/Host/Darwin/Mach.cpp

```diff
@@ -24,6 +24,39 @@
 #include <mach/thread_info.h>
 #include <sys/types.h>
 
+extern "C" {
```

Comment:
I don't understand what this block of code is trying to achieve. Can you put a comment explaining it?